standard-covoiturage_openapi.yaml

```diff
@@ -526,6 +554,113 @@ components:
           items:
             $ref: '#/components/schemas/JourneySchedule'
 
+    CarpoolBookingEvent:
+      required:
+        - id
+        - idToken
+        - data
+      properties:
+        id:
+          type: string
+          format: uuid
+          description:
+            Unique journey's id in the format of an [UUID](https://datatracker.ietf.org/doc/html/rfc4122).
+            Usage of a [4 UUID](https://datatracker.ietf.org/doc/html/rfc4122#section-4.4) generation algorithm
+            is advised.
+        idToken:
+          type: string
+          description: ID token of the user retrieved using the OpenID Connect flows.
+        data:
+          oneOf:
+          - $ref: '#/components/schemas/DriverCarpoolBooking'
+          - $ref: '#/components/schemas/PassengerCarpoolBooking'
+
+    CarpoolBooking:
+      required:
+      - id
+      - passengerPickupDate
+      - passengerPickupLat
+      - passengerPickupLng
+      - passengerDropLat
+      - passengerDropLng
+      - status
+      - webUrl
+      - price
+      properties:
+        id:
+          type: "string"
+          description: Unique identifier of the booking.
+        passengerPickupDate:
+          type: number
+          description: Passenger pickup datetime as a UNIX UTC timestamp in seconds.
+          format: long
+        passengerPickupLat:
+          type: "number"
+          format: "double"
+          description: Latitude of the passenger pick-up point.
+        passengerPickupLng:
+          type: "number"
+          format: "double"
+          description: Longitude of the passenger pick-up point.
+        passengerDropLat:
+          type: "number"
+          format: "double"
+          description: Latitude of the passenger drop-off point.
+        passengerDropLng:
+          type: "number"
+          format: "double"
+          description: Longitude of the passenger drop-off point.
+        passengerPickupAddress:
+          type: string
+          description: String representing the pickup-up address.
+        passengerDropAddress:
+          type: string
+          description: String representing the drop-off address.
+        status:
+          type: string
+          enum:
+            - WAITING_CONFIRMATION
+            - CONFIRMED
+            - CANCELLED
+            - COMPLETED_PENDING_VALIDATION
+            - VALIDATED
+        distance:
+          type: "integer"
+          description: |
+            Carpooling distance in meters. When the booking is COMPLETED or VALIDATED, this is the actual distance travelled if available.
+        duration:
+          type: integer
+          description: Carpooling duration in seconds.
+        webUrl:
+          type: string
+          description: URL of the booking on the webservice provider platform.
+
+    PassengerCarpoolBooking:
```

Comment:
What are the reasons to distinguish between the passenger and driver booking? Both should have a driver and at least one passenger (and by the way, should we handle multiple passengers here?).

Reply:
> What are the reasons to distinguish between the passenger and driver booking?

In this flow, the booking event is posted "in the name of the [driver/passenger]", authenticated through the `idToken`. So we only give here the info regarding the carpooling counterpart (passenger or driver). 

Proceeding like this ensures: 
- authentication: that there is no ambiguity for the MaaS platform on the identity of the user for which the booking should appear (the OpenID Connect `sub` is embeded in the `idToken`).
- authorization (GDPR compliance): tells the MaaS platform that the authenticated user has given its consent to share these booking data (which may not be the case for the counterpart).
- security (bonus): in case of credentials leaks, blast radius is limited to the leaked account credentials.

It also state clearly that the counterpart may not be known by the MaaS platform. Which is ok for this flow.

>  Both should have a driver and at least one passenger (and by the way, should we handle multiple passengers here?).

Regarding the possibility of multiple passengers for a booking, proceeding like this have been banned by IDFM and the RPC for good reasons: nothing is shared (itinerary, price, status) between multiple passengers except the identity of the driver. Trying to group multiple bookings for a driver in an API is only an added complexity. And the use-case can be solved using business rules (to group bookings) or a specific key (ex: `driver_carpooling_id`) but do we need/want that ?